panther_hardware_interfaces/include/panther_hardware_interfaces/panther_system_ros_interface.hpp

```diff
@@ -0,0 +1,115 @@
+// Copyright 2023 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef PANTHER_HARDWARE_INTERFACES_PANTHER_SYSTEM_ROS_INTERFACE_HPP_
+#define PANTHER_HARDWARE_INTERFACES_PANTHER_SYSTEM_ROS_INTERFACE_HPP_
+
+#include <memory>
+#include <thread>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <realtime_tools/realtime_publisher.h>
+
+#include <std_srvs/srv/trigger.hpp>
+
+#include <panther_msgs/msg/driver_state.hpp>
+
+#include <panther_hardware_interfaces/roboteq_data_converters.hpp>
+
+namespace panther_hardware_interfaces
+{
+
+struct CanErrors
+{
+  bool error;
+  bool write_sdo_error;
+  bool read_sdo_error;
+  bool read_pdo_error;
+  bool front_data_timed_out;
+  bool rear_data_timed_out;
+  bool front_can_net_err;
+  bool rear_can_net_err;
+};
```

Comment:
```suggestion
struct CanErrors
{
  const bool error;
  const bool write_sdo_error;
  const bool read_sdo_error;
  const bool read_pdo_error;
  const bool front_data_timed_out;
  const bool rear_data_timed_out;
  const bool front_can_net_err;
  const bool rear_can_net_err;
};
```
Open suggestion to make this structure immutable

Reply:
I would prefer to leave it mutable - I'm not sure if there is any specific reason why this structure should be immutable